utils/node-loader/src/batch_pool/generators/batch.rs

```diff
@@ -127,182 +147,91 @@ impl<Rng: CallGenRng> BatchGenerator<Rng> {
-        rt_settings: RuntimeSettings,
-    ) -> Batch {
-        let mut rng = Rng::seed_from_u64(seed);
-        let inner = utils::iterator_with_args(self.batch_size, || {
-            (existing_programs.clone(), rng.next_u64())
-        })
-        .enumerate()
-        .map(|(i, (existing_programs, rng_seed))| {
-            tracing::debug_span!("`send_message` generator", call_id = i + 1).in_scope(|| {
-                SendMessageArgs::generate::<Rng>(existing_programs, rng_seed, rt_settings.gas_limit)
-            })
-        })
-        .collect();
-
-        Batch::SendMessage(inner)
-    }
-
-    #[instrument(skip_all, fields(seed = seed, batch_type = "create_program"))]
-    fn gen_create_program_batch(
-        &mut self,
-        existing_codes: NonEmpty<CodeId>,
-        seed: Seed,
-        rt_settings: RuntimeSettings,
-    ) -> Batch {
-        let mut rng = Rng::seed_from_u64(seed);
-        let inner =
-            utils::iterator_with_args(self.batch_size, || (existing_codes.clone(), rng.next_u64()))
-                .enumerate()
-                .map(|(i, (existing_codes, rng_seed))| {
-                    tracing::debug_span!("`create_program` generator", call_id = i + 1).in_scope(
-                        || {
-                            CreateProgramArgs::generate::<Rng>(
-                                existing_codes,
-                                rng_seed,
-                                rt_settings.gas_limit,
-                            )
-                        },
-                    )
-                })
-                .collect();
-
-        Batch::CreateProgram(inner)
-    }
-
-    #[instrument(skip_all, fields(seed = seed, batch_type = "send_reply"))]
-    fn gen_send_reply_batch(
-        &mut self,
-        mailbox_messages: NonEmpty<MessageId>,
+    #[instrument(skip_all, fields(seed = seed, batch_type = T::name()))]
+    fn gen_batch<
+        T: Args + ArgsName,
+        FuzzerArgsFn: FnMut(&mut Rng) -> T::FuzzerArgs,
+        ConstArgsFn: Fn() -> T::ConstArgs,
+    >(
+        batch_size: usize,
         seed: Seed,
-        rt_settings: RuntimeSettings,
-    ) -> Batch {
+        mut fuzzer_args_fn: FuzzerArgsFn,
+        const_args_fn: ConstArgsFn,
+    ) -> Batch
+    where
+        Batch: From<Vec<T>>,
+    {
         let mut rng = Rng::seed_from_u64(seed);
-        let inner = utils::iterator_with_args(self.batch_size, || {
-            (mailbox_messages.clone(), rng.next_u64())
-        })
-        .enumerate()
-        .map(|(i, (mailbox_messages, rng_seed))| {
-            tracing::debug_span!("`send_reply` generator", call_id = i + 1).in_scope(|| {
-                SendReplyArgs::generate::<Rng>(mailbox_messages, rng_seed, rt_settings.gas_limit)
+        let inner: Vec<_> = utils::iterator_with_args(batch_size, || fuzzer_args_fn(&mut rng))
+            .enumerate()
+            .map(|(i, fuzzer_args)| {
+                tracing::debug_span!("generator", name = T::name(), call_id = i + 1)
```

Comment:
Try this for solving the problem stated in the PR head comment
```suggestion
                tracing::debug_span!(generator_for = T::name(), call_id = i + 1)
```

Reply:
or create a variable like `let name = T::name();` and use the `name` like it was in the previous code.

Reply:
like
```rust
let name = T::name();
tracing::debug_span!("{} generator", name, call_id = i + 1)
```

Reply:
what about that
```rust
let name = T::name();
tracing::debug_span!(generator_for = name, call_id = i + 1)
```

or the other one up there in the comment?